### PR TITLE
[EIP-3742] - Standard for multi party contract

### PR DESCRIPTION
**Pull request by saurabhsanthosh**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
See [EIP-1](https://eips.ethereum.org/EIPS/eip-1) and https://github.com/ethereum/EIPs/blob/master/eip-template.md for details on how to construct an EIP.

**Comment by saurabhsanthosh** [comment]
> See [EIP-1](https://eips.ethereum.org/EIPS/eip-1) and https://github.com/ethereum/EIPs/blob/master/eip-template.md for details on how to construct an EIP.

I have made following changes :
Left the EIP number blank
Change the initial information according to template

What else i have to change? Thanks a lot for the help

**Comment by saurabhsanthosh** [comment]
PFA the implementation and tests

Implementation : https://github.com/saurabhsanthosh/ethereum_EIPs/blob/master/contracts/erc27/ERC27.sol
Tests : https://github.com/saurabhsanthosh/ethereum_EIPs/blob/master/test/erc27.js

**Comment by saurabhsanthosh** [comment]
@MicahZoltu Can you help me with this? It would be great if you can quickly point out what all I am missing. Thanks :)

**Comment by MicahZoltu** [comment]
I generally don't review ERCs.  Someone else will eventually get around to doing a review and helping you through the process, though I don't know what the availability of other editors is like these days.

File: EIPS/eip-multi_party_contract.md
Comment (MicahZoltu) [inline EIPS/eip-multi_party_contract.md:2]:
```suggestion
eip: 3742
```

File: EIPS/eip-multi_party_contract.md
Comment (MicahZoltu) [inline EIPS/eip-multi_party_contract.md:4]:
```suggestion
author: Saurabh Santhosh (@saurabhsanthosh), Saurabh Santhosh <saurabhsanthosh@gmail.com>
```
or
```suggestion
author: Saurabh Santhosh (@saurabhsanthosh)
```

At least one author should be a GitHub handle.

**Comment by saurabhsanthosh** [comment]
> I generally don't review ERCs. Someone else will eventually get around to doing a review and helping you through the process, though I don't know what the availability of other editors is like these days.

Thanks a lot for the help. Have made the suggested changes. Hope someone will start reviewing the PR soon.

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:12]: We recently deprecated the "Simple Summary" in favor of a description field in the preamble. Please see the [EIP template ](https://github.com/ethereum/EIPs/blob/master/eip-template.md)for an example.

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:310]: Please remove this section, it isn't a valid EIP-1 section

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:20]:
```suggestion
The following standard allows for the implementation of a standard API for co-owning assets within smart contracts. This standard provides basic functionality to add members, remove members and create/execute contract transactions approved by all members of the contract. Some of the possible actions are token transferring, operator approvals and member addition/removals. It's important to note that every group of users will be a new instance of the contract having a unique address.
```

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:1]: You will need to add a `discussions-to` link for your EIP to be merged as a draft.

**Review by lightclient — CHANGES_REQUESTED** [review]
Could you share how your EIP is different than a multi-signature wallet?

File: EIPS/eip-multi_party_contract.md
Comment (saurabhsanthosh) [reply EIPS/eip-multi_party_contract.md:1]: Added the discussions-to

**Comment by saurabhsanthosh** [comment]
> Could you share how your EIP is different than a multi-signature wallet?

First of all Thanks a lot for reviewing this PR. @lightclient 

The contract standard proposed by this EIP is mainly for managing smart contract by a group of people in a secure and democratic manner. Holding tokens, ETH is just one of the use cases (which comes under general umbrella of management)

Eg : Following are few of the use cases which this EIP can address which a multi-signature wallet cannot.

1). We have a smart contract managed by 3 people and we have to distribute assets acquired by the smart contract to all members. With this EIP standard all we have to do is define an action like below 

Action :
transfer(address1, total/3)
transfer(address2, total/3)
transfer(address3, total/3)

This Action will be executed in an atomic way (Either all 3 transaction succeed or none of them does) after the approval of all the members. 

2). Paying off a member of the group. We have to remove a member by giving him//her some incentive

With this EIP standard all we have to do is define an action like below 

Action :
transfer(address1, <some-asset>)
removeMember(address1)

This Action will be approved even by the member being removed.

3). 2 People have to transfer their NFTs to a smart contract which they co-manage. In this case the transfers have to take place atomically to protect both parties. In this scenario action can be like following

Action :
transferERC721(address1, <contract_address>, <tokenId1>)
transferERC721(address2, <contract_address>, <tokenId2>)

Above mentioned are just few of the examples. With the generic notion of Actions (each capable of executing multiple transactions in atomic way) and democratic way of approving/executing these actions can solve wide range of complex use cases in multi party contracts.

Please let me know if you have any. more queries.

**Comment by MicahZoltu** [comment]
There are many EIPs that shouldn't be EIPs in my opinion, but we (editors) try to only guide rather than assert what is and is not allowed.  Technically, you can write a valid EIP on how to make a sandwich.  I would advise against doing such a thing and encourage anyone who wanted to describe how to make a sandwich to do so elsewhere, but that doesn't mean I would outright reject the EIP.  😊

**Comment by saurabhsanthosh** [comment]
> There are many EIPs that shouldn't be EIPs in my opinion, but we (editors) try to only guide rather than assert what is and is not allowed. Technically, you can write a valid EIP on how to make a sandwich. I would advise against doing such a thing and encourage anyone who wanted to describe how to make a sandwich to do so elsewhere, but that doesn't mean I would outright reject the EIP. 😊

Rightly said! Thanks a lot for taking the time to answer the questions and reviewing this PR. Appreciate it :)

**Comment by lightclient** [comment]
@saurabhsanthosh: I also agree with @MicahZoltu here. The use cases you've laid out are already possible with mutlsig wallets today. Additionally, when making an ERC it's critical to consider how important it is for everyone to use the same interface. In the case of this (and multisigs in general), it isn't (yet) critical that there is a standard multisig interface. I can imagine a world were such a standard would be useful (e.g. many-to-many relationship of multisig smart contracts to client software).

We generally don't want to outright block an EIP, so please let me know if you wish to continue pursuing this and I will merge once it meets the minimum requirements for a draft.

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:17]:
```suggestion
Please remove the extraneous spacing in this document.
```

File: EIPS/eip-multi_party_contract.md
Comment (lightclient) [inline EIPS/eip-multi_party_contract.md:289]:
```suggestion
## Reference Implementation
```

**Review by lightclient — CHANGES_REQUESTED** [review]
Missing `Rationale` and `Security Consideration` sections.

**Pull request closed without merging**
